### PR TITLE
JAMES-3946 Add a DropLists postgresql backend

### PR DESCRIPTION
**Pull request by Maxxx873**

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]
Looks good to me, thanks!

File: server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropList.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            .and(DENIED_ENTITY.eq(entry.getDeniedEntity()))));
+    }
+
+    @Override
+    public Flux<DropListEntry> list(OwnerScope ownerScope, String owner) {
+        Preconditions.checkArgument(ownerScope != null);
+        Preconditions.checkArgument(owner != null);
+        return postgresExecutor.executeRows(dsl -> Flux.from(dsl.selectFrom(TABLE_NAME)
+                .where(OWNER_SCOPE.eq(ownerScope.name()))
+                .and(OWNER.eq(owner))))
```
Comment (quantranhong1999) [inline server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropList.java:89]: We should add an index for (OWNER_SCOPE, OWNER) IMO.

Maybe another one for (OWNER_SCOPE, OWNER, DENIED_ENTITY)

File: server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropListModule.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        PostgresTable TABLE = PostgresTable.name(TABLE_NAME.getName())
+            .createTableStep(((dsl, tableName) -> dsl.createTableIfNotExists(tableName)
+                .column(DROPLIST_ID)
+                .column(OWNER_SCOPE)
+                .column(OWNER)
+                .column(DENIED_ENTITY_TYPE)
+                .column(DENIED_ENTITY)
+                .constraint(DSL.primaryKey(DROPLIST_ID))))
+            .supportsRowLevelSecurity()
```
Comment (quantranhong1999) [inline server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropListModule.java:50]: We likely should disable RLS here (can we extract the domain (RLS column) from the owner input?)

File: server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropList.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresDropList implements DropList {
+    private final PostgresExecutor postgresExecutor;
+
+    @Inject
+    public PostgresDropList(@Named(DEFAULT_INJECT) PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
```
Comment (quantranhong1999) [inline server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropList.java:56]: If we enable RLS, we can not use the `@Named(DEFAULT_INJECT) PostgresExecutor postgresExecutor`.

File: server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropList.java (hunk 1 above)
Comment (Maxxx873) [reply server/data/data-postgres/src/main/java/org/apache/james/droplists/postgres/PostgresDropList.java:89]: thank you for your valuable comments!

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Maxxx873** [comment]
Can we merge this or does anything need to be improved?

**Comment by chibenwa** [comment]
It would be wonderful to get the CI to validate the latest diff.

I relaunched a build.

**Review by quantranhong1999 — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by chibenwa** [comment]
And here we go!

**Pull request merged**